### ISSUE TITLE
Raw syntax for lifetimes and loop labels

### ISSUE THREAD

**Issue by traviscross** [rust-lang/rust#126335]
We don't currently accept:

```rust
pub fn label() {
    'r#async: loop {
        break 'r#async;
    }
}

pub fn lifetime<'r#async>() {}
```

Should we?  This has relevance to edition migrations.

See:

- https://github.com/rust-lang/rust/issues/125986

**Comment by traviscross** [comment]
@rustbot labels -I-lang-nominated

We discussed this in the lang triage call today.  Our consensus was that this should be allowed and that it seems to have been an oversight that it wasn't already under [RFC 3101](https://github.com/rust-lang/rfcs/pull/3101) and other related work.

**Comment by compiler-errors** [comment]
I'll put up a PR for this tomorrow. I've got a draft, just want to make sure it's actually correct.

**Comment by traviscross** [comment]
This has been fixed by [#126452](https://github.com/rust-lang/rust/pull/126452), so we can close this.
